subgraphs/rocket-pool/src/mappings/RocketPool.ts

```diff
@@ -0,0 +1,251 @@
+import {
+  RPLStaked,
+  RPLSlashed,
+  RPLWithdrawn,
+} from "../../generated/rocketNodeStaking/rocketNodeStaking";
+import {
+  MinipoolEnqueued,
+  MinipoolRemoved,
+  MinipoolDequeued,
+} from "../../generated/rocketMinipoolQueue/rocketMinipoolQueue";
+
+import { BalancesUpdated } from "../../generated/rocketNetworkBalances/rocketNetworkBalances";
+
+import { rocketNetworkPrices } from "../../generated/rocketNodeStaking/rocketNetworkPrices";
+import { ERC20 } from "../../generated/rocketVault/ERC20";
+import { getOrCreateToken } from "../entities/token";
+
+import { updateUsageMetrics } from "../entityUpdates/usageMetrics";
+import {
+  updateProtocolAndPoolTvl,
+  updateSnapshotsTvl,
+  updateSupplySideRevenueMetrics,
+  updateProtocolSideRevenueMetrics,
+  updateTotalRevenueMetrics,
+  updateMinipoolTvlandRevenue,
+} from "../entityUpdates/financialMetrics";
+import {
+  ZERO_ADDRESS,
+  ETH_ADDRESS,
+  BIGINT_ZERO,
+  RPL_ADDRESS,
+  RETH_ADDRESS,
+  BIGINT_NEGATIVE_ONE,
+  BIGDECIMAL_HALF,
+  BIGINT_TEN_TO_EIGHTEENTH,
+  PRICEENCODE,
+  NODEDEPOSIT_ENCODE,
+  STORAGE,
+  ONE_ETH_IN_WEI,
+  BIGDECIMAL_ONE,
+  BIGDECIMAL_ZERO,
+} from "../utils/constants";
+import { getOrCreatePool } from "../entities/pool";
+import { getOrCreateProtocol } from "../entities/protocol";
+import { getOrCreateMinipool } from "../entities/minipool";
+
+/** Queries Storage Address for current address of encode. encode is the string of the name of the rocketpool contract, which has been keccack256(abi.unpack(string)) in solidity. */
+export function getStorageAddress(encode: Bytes): Address {
+  const storage = RocketStorage.bind(Address.fromString(STORAGE));
+  let address = storage.try_getAddress(encode);
+  if (address.reverted) {
+    log.info("getStorageAddress call reverted", []);
+    return Address.fromString("0x0000000000000000000000000000000000000000");
+  } else {
+    return Address.fromBytes(address.value);
+  }
+}
+
+/** handleEtherDeposit tracks ether deposited into rocketpool, which represents the TVL of the ETH staked in the pool. */
+export function handleEtherDeposit(event: EtherDeposited): void {
+  updateUsageMetrics(event.block, event.transaction.from);
+  updateProtocolAndPoolTvl(event.block, event.params.amount, BIGINT_ZERO);
+  updateSnapshotsTvl(event.block);
+}
+
+/** handleEtherWithdrawn tracks ether withdrawn into rocketpool, which represents the TVL of the ETH staked in the pool. */
+
+export function handleEtherWithdrawn(event: EtherWithdrawn): void {
+  updateUsageMetrics(event.block, event.transaction.from);
+  updateProtocolAndPoolTvl(
+    event.block,
+    BIGINT_NEGATIVE_ONE.times(event.params.amount),
+    BIGINT_ZERO
+  );
+  updateSnapshotsTvl(event.block);
+}
+
+/** Can potentially use minipool created/destroyed to track this instead */
+
+/** Staking rewards is a function of amt of reth/eth ratio so can be calculated for any minipool based on amt of staked eth, and then slashing is subtracted from it ? */
```

Comment:
Is this a question? :P

Reply:
Just my manic scratchwork. Will delete XD